src/mainwindow.cpp

```diff
@@ -892,16 +892,51 @@ bool MainWindow::event(QEvent *event)
     {
         if (m_dropMode &&
             !Properties::Instance()->dropKeepOpen &&
-            qApp->activeWindow() == nullptr
-            )
-            hide();
+            qApp->activeWindow() == nullptr)
+        {
+            // On Wayland and with a modal dialog, the dropdown window can be activated by
+            // clicking inside it and then deactivated by clicking on another window (see below).
+            if (!m_layerWindow || m_layerWindow->layer() == LayerShellQt::Window::Layer::LayerOverlay)
+            {
+                hide();
+            }
+        }
     }
+    // A workaround for the modal dialogs of the dropdown window on Wayland.
+    else if (event->type() == QEvent::WindowBlocked && m_layerWindow && m_dropMode)
+    {
+        if (auto dialog = qobject_cast<QDialog*>(qApp->activeModalWidget()))
+        {
+            dialog->winId();
+            if (QWindow *win = dialog->windowHandle())
+            {
+                if (LayerShellQt::Window *layershell = LayerShellQt::Window::get(win))
+                {
+                    layershell->setLayer(LayerShellQt::Window::Layer::LayerOverlay);
+                    layershell->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityOnDemand);
+                    LayerShellQt::Window::Anchors anchors = {LayerShellQt::Window::AnchorTop};
+                    layershell->setAnchors(anchors);
+                    layershell->setScreenConfiguration(LayerShellQt::Window::ScreenConfiguration::ScreenFromCompositor);
+                    // For preventing the dialog from going behind the dropdown window if
+                    // the latter is clicked, the window is set on the top layer temporarily.
+                    // It will be set on the overlay layer again when the dialog exits.
+                    m_layerWindow->setLayer(LayerShellQt::Window::Layer::LayerTop);
```

Comment:
I'm not sure if this really works, I see 2 overlay layers and no top layer:
```
        Layer level 2 (top):
                Layer 644d83fae6a0: xywh: 0 0 1920 48, namespace: dock
                Layer 644d8274d410: xywh: 1877 203 43 723, namespace: dock
                Layer 644d82773c90: xywh: 163 48 1551 935, namespace: dropdown_terminal
        Layer level 3 (overlay):
                Layer 644d83f7bc20: xywh: 805 48 268 138, namespace: window

```
Some compositors put the new layer under the existing one (sway I guess).